Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp

```diff
@@ -141,320 +120,474 @@ namespace AzToolsFramework
-                            watcher->GetCommunicator()->ReadOutput(consoleBuffer.data(), outputSize);
-                            consoleOutput += consoleBuffer;
-                        }
-                    }
+                    int result = archive->UpdateFile(
+                        filePathLine, fileBuffer.data(), fileBuffer.size(), s_compressionMethod,
+                        s_compressionLevel, s_compressionCodec);
+
+                    bool thisSuccess = (result == AZ::IO::ZipDir::ZD_ERROR_SUCCESS);
+                    success = (success && thisSuccess);
+                    AZ_Error(
+                        s_traceName, thisSuccess, "Error %d encountered while adding '%.*s' to archive '%s'", result,
+                        AZ_STRING_ARG(filePathLine), archive->GetFullPath());
                 }
                 else
                 {
-                    ConsoleEchoCommunicator echoCommunicator(watcher->GetCommunicator());
-                    while (watcher->IsProcessRunning(&exitCode))
-                    {
-                        if (!taskHandle.IsNull())
-                        {
-                            AZStd::unique_lock<AZStd::mutex> lock(m_threadControlMutex);
-                            if (m_threadInfoMap[taskHandle].shouldStop)
-                            {
-                                watcher->TerminateProcess(static_cast<AZ::u32>(SevenZipExitCode::UserStoppedProcess));
-                            }
-                        }
-                        watcher->WaitForProcessToExit(g_sleepDuration, &exitCode);
-                        echoCommunicator.Pump();
-                    }
+                    AZ_Error(
+                        s_traceName, false, "Error encountered while reading '%.*s' to add to archive '%.*s'", AZ_STRING_ARG(filePathLine),
+                        AZ_STRING_ARG(archive->GetFullPath().Native()));
                 }
-            }
+            };
 
-            if (taskHandle.IsNull())
-            {
-                respCallback(exitCode == static_cast<AZ::u32>(SevenZipExitCode::NoError), AZStd::move(consoleOutput));
-            }
-            else
+            ArchiveUtils::ProcessFileList(listFilePath, PerLineCallback);
+
+            archive.reset();
+            p.set_value(success);
+        };
+
+        // Async task...
+        std::promise<bool> p;
+        std::future<bool> f = p.get_future();
+
+        AZStd::thread_desc threadDesc;
+        threadDesc.m_name = "Archive Task (Add)";
+        m_threads.emplace_back(threadDesc, FnAddFilesToArchive, AZStd::move(p));
+        return AZStd::move(f);
+    }
+
+
+    bool ArchiveComponent::CheckParamsForAdd(const AZStd::string& directory, const AZStd::string& file)
+    {
+        if (!m_localFileIO || !m_archive)
+        {
+            return false;
+        }
+
+        if (!m_localFileIO->Exists(directory.c_str()) || !m_localFileIO->IsDirectory(directory.c_str()))
```

Comment:
The IsDirectory check subsumes the Exists check.
If the working directory is a "directory", then it exists.

So the Exists check isn't needed.

```suggestion
        if (!m_localFileIO->IsDirectory(directory.c_str()))
```